### PR TITLE
schema with key or id already exists

### LINKED ISSUE #5772: 4.x schema with key or id "${id}" already exists
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Hello,

I exported a contract ABI through bscscan, for contract address 0x66665ca5cb0f83e9cb813e89ca64bd6cdd4c6666
When using it locally, it immediately triggers the following error message:
```
Error: schema with key or id "1066c1c1e9545479d0a6dacf687763315a77dec942e449ac2d73f088f8f1b391541879a1b56ca748e0dc06fe69bfb8bb7a0920c63e5a93baf5563150735d4271" already exists
    at Ajv._checkUnique (../node_modules/ajv/lib/core.ts:727:13)
    at Ajv._addSchema (../node_modules/ajv/lib/core.ts:718:24)
    at Ajv.addSchema (../node_modules/ajv/lib/core.ts:486:30)
    at Web3Validator.validate (../node_modules/web3-validator/src/web3_validator.ts:105:20)
    at Object.<anonymous> (../node_modules/web3-eth-contract/src/contract.ts:1028:15)
    at Stats.<anonymous> (../src/classes/projects/Ark/Actions/Stats.ts:25:55)
    at Generator.next (<anonymous>)
```

### Expected Behavior

I expect to receive the result of calling methods on this contract.
The same code works with ABIs of other contracts, but never on any methods of this contract.

### Steps to Reproduce

Using this code:
```typescript
// @ts-ignore
const balance:any = await this.contract.methods.principalBalance(this.web3.getUserWalletAddress()).call(); // this.contract is assigned with the return from new this.web3JS.eth.Contract(ABI, address);
console.log(balance);
```
Here is the contract ABI: [0x66665ca5cb0f83e9cb813e89ca64bd6cdd4c6666.zip](https://github.com/web3/web3.js/files/10472773/0x66665ca5cb0f83e9cb813e89ca64bd6cdd4c6666.zip)

### Web3.js Version

4.0.1-alpha.5

### Environment

Operating System: Ubuntu 20.04 (through Win 11 Pro / WSL2)
Browser: Microsoft Edge Version 108.0.1462.54
Node.js Version: v18.10.0
NPM Version: 9.3.1 (using yarn 1.22.19)

### Anything Else?

To extract the ABI, I used this link: https://bscscan.com/address/0x66665ca5cb0f83e9cb813e89ca64bd6cdd4c6666#code
As for package version (4.x vs 1.x), I'm running this tool locally for fun and (my own) education purposes, it's not meant for production, I'll be happy to be your guinea pig on this one, if necessary

### PR DESCRIPTION
**Pull request by nikoulai**
## Description

closes #5772 

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../blob/1.x/CONTRIBUTIONS.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist for 1.x:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

**Review by avkos — CHANGES_REQUESTED** [review]
coverage decreased. please cover changes with unit tests

File: packages/web3-validator/src/web3_validator.ts [hunk 1]
```diff
@@ -101,7 +101,19 @@ export class Web3Validator {
...
+
 		if (!this._validator.getSchema(schemaKey)) {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+			if (jsonSchema?.items && jsonSchema.items.length > 0) {
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, no-plusplus
+				for (let i = 0; i < jsonSchema.items.length; i++) {
+					// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+					if (jsonSchema?.items[i]?.$id === '') {
+						// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+						jsonSchema.items[i].$id = `${schemaKey}/items/${i}`;
```
Comment (Muhammad-Altabba) [inline packages/web3-validator/src/web3_validator.ts:113]: With something like the following, we can use `eslint-disable-next-line` only once:
```suggestion
			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
			const jsonSchemaItems = jsonSchema?.items;
			if (jsonSchemaItems && jsonSchemaItems.length > 0) {
				for (let i = 0; i < jsonSchemaItems.length; i++) {
					if (jsonSchemaItems[i]?.$id === '') {
						jsonSchemaItems[i].$id = `${schemaKey}/items/${i}`;
```

**Review by Muhammad-Altabba — COMMENTED** [review]
Looks good except for not having tests as highlighted by @avkos.
Please link the PR to its issue.

File: packages/web3-validator/src/web3_validator.ts (hunk 1 above)
Comment (nikoulai) [reply packages/web3-validator/src/web3_validator.ts:113]: thanks for the suggestion, Muhammad!

**Review by luu-alex — APPROVED** [review]

**Review by avkos — APPROVED** [review]

File: packages/web3-validator/src/web3_validator.ts (hunk 1 above)
Comment (Muhammad-Altabba) [reply packages/web3-validator/src/web3_validator.ts:113]: You are welcome @nikoulai ,
However, I see that the code was not updated. Did you forget to push a local code?

Comment (nikoulai) [reply packages/web3-validator/src/web3_validator.ts:113]: I made the change and reverted due to failure in the pipeline. Please see the action: https://github.com/web3/web3.js/actions/runs/4787005578/jobs/8511913375. If I misunderstood something I can open a small PR for the suggestion.

Comment (Muhammad-Altabba) [reply packages/web3-validator/src/web3_validator.ts:113]: The idea is to minimize the number of ignores by having an assigned variable once. Here is a working snippet:

```suggestion
			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
			const jsonSchemaItems = jsonSchema?.items as { $id: string }[];
			if (jsonSchemaItems && jsonSchemaItems.length > 0) {
				// eslint-disable-next-line no-plusplus
				for (let i = 0; i < jsonSchemaItems.length; i++) {
					if (jsonSchemaItems[i]?.$id === '') {
						jsonSchemaItems[i].$id = `${schemaKey}/items/${i}`;
```

You may check:
https://github.com/web3/web3.js/commit/51772eb81134641cb41b5727d9058c1b241ebd87

**Pull request merged**